### ISSUE TITLE
Applications with a "Pending" status should not be editable by industrial users

### ISSUE THREAD

**Issue by andrea-williams** [bcgov/cas-registration#285]
#### Description:

Applications (for both operators and operations) should not be editable when their status = "Pending". PR in progress you can continue if it's helpful: https://github.com/bcgov/cas-registration/pull/423

#### Acceptance Criteria:

Given an operation has a status of "Pending"
When an industrial user views the form data for their pending operation
Then all the data fields are view-only (can't be edited)

Given an operator has a status of "Pending"
When an industrial user views the form data for their pending operator
Then all the data fields are view-only (can't be edited)

NOTE: I'm guessing that CAS staff should never be able to edit operators/operations directly - needs to be confirmed. UPDATE Nov 27: Hanna Confirmed with business area: CAS staff should not be able to edit operators/operations directly 

#### Dependencies
- Blocked by #25 

#### Development Checklist:

- [ ] readOnly flag on Operator and Operations forms is set depending on (a) assigned role of user AND (b) whether the application's `status == PENDING`. 
- [ ] if status == "Not Registered", form should be editable by industrial users with the appropriate permissions.
- [ ] if Operation or Operator status == "Pending", it is not editable by industrial users
- [ ] if Operation or Operator status == "Approved", it is editable
- [ ] if Operation or Operator status == "Rejected", it is editable by industrial users
- [ ] if Operation or Operator status == "Changes Requested", it is editable by industrial users
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by Dianadec** [comment]
@hannavovk will confirm this question in the description with the business area ("I'm guessing that CAS staff should never be able to edit operators/operations directly - needs to be confirmed.")
@andrea-williams outside of that, could a development checklist be added to this card at this point so that we can refine it and estimate it?

**Comment by andrea-williams** [comment]
Added details to development checklist, although there are still some outstanding questions for either Hanna or the business area.

**Comment by BCerki** [comment]
Is this just about making the form readonly on the frontend, or does it include some sort of db locking?

**Comment by Dianadec** [comment]
@andrea-williams  do you have the answer to Brianna's question? 

> Is this just about making the form readonly on the frontend, or does it include some sort of db locking?

**Comment by andrea-williams** [comment]
@BCerki @Dianadec  just making the form readonly on the frontend :)

**Comment by Dianadec** [comment]
Reminder to provide your estimates please: @andrea-williams @dleard @BCerki @marcellmueller @Sepehr-Sobhani

**Comment by BCerki** [comment]
Started a PR before realizing this was blocked: https://github.com/bcgov/cas-registration/pull/423

**Comment by BCerki** [comment]
Will implementing this fix this bug? https://github.com/bcgov/cas-registration/issues/486

**Comment by Dianadec** [comment]
Might need some changes once #441 goes in

**Comment by marcellmueller** [comment]
@BCerki do you think we can delete this issue? The Operations form already has editing disabled for `PENDING` status plus 600 just shows the Request message until a User Operator is approved.

**Comment by BCerki** [comment]
> @BCerki do you think we can delete this issue? The Operations form already has editing disabled for `PENDING` status plus since #600 was merged the User Operator just shows the Request message until they are approved.

Yup, I think we're good to close
